syntax="proto3";

import "common.proto";
package MSG;


// keep alive 
message PingReq {
	string sessionKey = 1;
}

message PongAns {
	ErrorCode err = 1;	
}


// login request
message LoginReq {
	string name = 1;
}

// login answer
message LoginAns {
	ErrorCode err = 1;
	string sessionKey = 2;
	UserData_ data = 3;
}


// regist request
message RegistReq {
	UserData_ data = 1;
}

// regist answer
message RegistAns {
	ErrorCode err = 1;
}

// version
message VersionReq {
	string version = 1;
}

message VersionAns {
	ErrorCode err = 1;
	// available contents list
	repeated Contents_ contents = 2;
}
